src/inputfield.js

```diff
@@ -1,34 +1,58 @@
-import React, {Component} from 'react'
+import React, { Component } from 'react'
+import PropTypes from 'prop-types';
+import omit from 'omit.js'
 
-// detect whether it is chrome 
+// detect whether it is chrome
 const isChrome = !!window.chrome && !!window.chrome.webstore
 
+const noop = () => {}
+
 class InputField extends Component {
+
+    static propTypes = {
+        value: PropTypes.any,
+        defaultValue: PropTypes.any,
+        onChange: PropTypes.func,
+        onInputChange: PropTypes.func
+    }
+
+    static defaultProps = {
+        onChange: noop,
+        onInputChange: noop
+    }
+
     constructor(props, context) {
         super(props, context)
         this.state = {
-            tempInput: this.props.defaultValue || '',
+            value: this.props.value || this.props.defaultValue || ''
         }
         this.isOnComposition = false
         this.emittedInput = true
     }
 
+    componentWillReceiveProps(nextProps) {
+        const { value } = nextProps
+        if (value !== this.props.value) {
+            this.setState({
```

Comment:
Nice. Maybe we need to initialize `isOnComposition` & `emittedInput` here.
```
this.isOnComposition = false
this.emittedInput = true
```

Reply:
It seems we cannot initialize `isOnComposition` & `emittedInput` in `componentWillReceiveProps` lifecycle method, otherwise the values will always be initialized when typing, and cannot handle composition event.

Reply:
I mean to change it when we receive new props. That's fine. I'll accept the  PR and add it myself.